src/main/java/com/google/cloud/teleport/newrelic/utils/ConfigHelper.java

```diff
@@ -15,9 +21,18 @@ private ConfigHelper() {
      * @param <T> The type of the value being read
      * @return The value included in the provided ValueProvider, if it's available and non-null, otherwise the default value.
      */
-    public static <T> T valueOrDefault(ValueProvider<T> value, T defaultValue ) {
+    public static <T> ValueProvider<T> valueOrDefault(ValueProvider<T> value, T defaultValue ) {
         return (value != null && value.isAccessible()) && value.get() != null
-                ? value.get()
-                : defaultValue;
+                ? StaticValueProvider.of(value.get())
+                : StaticValueProvider.of(defaultValue);
+    }
+
+    public static void logConfiguration(final NewRelicConfig newRelicConfig) {
```

Comment:
❓ it is not possible for `slf4j` to log objects?? I feel like this class may not be justified enough to exist 🤔

Reply:
Addressed this and switched to `toString`.